### DISCUSSION TITLE
Best practice untuk menggunakan read replicas RDS

### DISCUSSION THREAD

**Discussion by Kyuubang** [awsugid/forum#22] [Q&A] [answered]
Hi all, aku mau tanya dibeberapa case kita sering ditemui dengan traffic tinggi dimana ada keterbatasan iops atau bandwidth database (biasanya Read operations). apakah dengan menggunakan read replicas dapat mangatasi hal tersebut? lalu bagaimana baiknya kita managed traffic app read/write ke db? apakah menggunakan abstraksi db di app atau proxysql untuk management read and write query? atau apakah ada best practice untuk kasus ini? 

Terima kasih.

**Comment by rioastamal** [comment, marked as answer]
@Kyuubang. Yes, menggunakan read replica tentu membantu karena proses read akan displit ke beberapa node replica bukan hanya satu (master). Jika ingin garansi IOPS bisa menggunakan RDS dengan tipe [Provisioned IOPS SSD storage](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Storage.html), namun untuk aplikasi secara umum replika sudah cukup membantu.

Untuk masalah read dan write-nya bisa diatur disisi aplikasi. Ketika ada proses write/transtaction maka larinya ke master node tapi untuk read (SELECT) larinya ke endpoint replica.

Proxysql berguna untuk melakukan pooling connection sehingga koneksi yang dibuka ke database tidak terlalu banyak karena aplikasi akan menggunakan pool yang sudah ada. Ini berguna jika aplikasi memang traffic-nya sangat tinggi sehingga mengurangi beban koneksi yang dibuka di sisi database server.
